src/pull_request.ts

```diff
@@ -94,4 +96,106 @@ export async function handlePullRequest(config: Config) {
     body: buildWalkthroughMessage(summary),
   });
   info(`updated overview comment with walkthrough`);
+
+  // Review PR code changes
+  const review = await runReviewPrompt({
+    files: fileDiffs,
+    prTitle: pull_request.title,
+    prDescription: pull_request.body || "",
+    prSummary: summary.description,
+  });
+  info(`reviewed pull request`);
+
+  // Post review comments
+  await submitReview(
+    octokit,
+    context,
+    {
+      number: pull_request.number,
+      headSha: pull_request.head.sha,
+    },
+    review.comments
+  );
+  info(`posted review comments`);
+}
+
+async function submitReview(
+  octokit: Octokit,
+  context: Context,
+  pull_request: {
+    number: number;
+    headSha: string;
+  },
+  comments: AIComment[]
+) {
+  const submitInlineComment = async (
+    file: string,
+    line: number,
+    content: string
+  ) => {
+    await octokit.pulls.createReviewComment({
+      ...context.repo,
+      pull_number: pull_request.number,
+      commit_id: pull_request.headSha,
+      path: file,
+      body: content,
+      line,
+    });
+  };
+
+  // Handle file comments
+  const fileComments = comments.filter((c) => !c.end_line);
+  if (fileComments.length > 0) {
+    const responses = await Promise.allSettled(
+      fileComments.map((c) => submitInlineComment(c.file, -1, c.content))
+    );
+
+    for (const response of responses) {
+      if (response.status === "rejected") {
+        warning(`error creating file comment: ${response.reason}`);
+      }
+    }
+  }
+
+  // Handle line comments
+  const lineComments = comments.filter((c) => !!c.end_line);
+
+  // Try to submit all comments at once
+  try {
+    let commentsData = lineComments.map((c) => ({
+      path: c.file,
+      body: c.content,
+      line: c.end_line,
+      side: "RIGHT",
+      start_line:
+        c.start_line && c.start_line < c.end_line ? c.start_line : undefined,
+      start_side:
+        c.start_line && c.start_line < c.end_line ? "RIGHT" : undefined,
+    }));
+
+    const review = await octokit.pulls.createReview({
+      ...context.repo,
+      pull_number: pull_request.number,
+      commit_id: pull_request.headSha,
+      comments: commentsData,
+    });
+
+    await octokit.pulls.submitReview({
+      ...context.repo,
+      pull_number: pull_request.number,
+      review_id: review.data.id,
+      event: "COMMENT",
+      body: "Review submitted",
+    });
+  } catch (error) {
+    warning(`error submitting review: ${error}`);
+
+    // If submitting all comments at once fails, try submitting them one by one
+    info("trying to submit comments one by one");
+    await Promise.allSettled(
+      lineComments.map((c) =>
+        submitInlineComment(c.file, c.end_line, c.content)
+      )
+    );
```

Comment:
The fallback error handling doesn't log individual comment submission failures. Consider adding error logging for each failed comment submission to help with debugging:
```typescript
const results = await Promise.allSettled(
  lineComments.map((c) =>
    submitInlineComment(c.file, c.end_line, c.content)
  )
);
results.forEach((result, index) => {
  if (result.status === 'rejected') {
    warning(`Failed to submit comment for ${lineComments[index].file}: ${result.reason}`);
  }
});
```